apps/remote_control/test/lexical/remote_control/code_intelligence/definition_test.exs

```diff
@@ -0,0 +1,318 @@
+      uses_content = ~q[
+        defmodule UsesOwnFunction do
+          def greet do
+          end
+
+          def uses_greet do
+            greet|()
+          end
+        end
+      ]
+
+      {:ok, referenced_uri, definition_line} = definition(project, uses_content)
+
+      assert definition_line == ~S[  def «greet» do]
+      assert referenced_uri =~ "navigations/lib/my_module.ex"
+    end
+
+    test "find the attribute", %{project: project} do
+      uses_content = ~q[
+        defmodule UsesAttribute do
+          @b 2
+
+          def use_attribute do
+            @|b
+          end
+        end
+        ]
+      {:ok, referenced_uri, definition_line} = definition(project, uses_content)
+
+      assert definition_line =~ ~S[«@b» 2]
+      assert referenced_uri =~ "navigations/lib/my_module.ex"
+    end
+
+    test "find the variable", %{project: project} do
+      uses_content = ~q[
+        defmodule UsesVariable do
+          def use_variable do
+            a = 1
+
+            if true do
+              a|
+            end
+          end
+        end
+      ]
+      {:ok, referenced_uri, definition_line} = definition(project, uses_content)
+
+      assert definition_line =~ ~S[«a» = 1]
+      assert referenced_uri =~ "navigations/lib/my_module.ex"
+    end
+
+    test "can't find the definition when call a Elixir std module function",
+         %{project: project} do
+      uses_content = ~q[
+        String.to_integer|("1")
+      ]
+
+      # credo:disable-for-next-line Credo.Check.Design.TagTODO
+      # TODO: this should be fixed when we have a call tracer
+      {:ok, nil} = definition(project, uses_content)
+    end
+
+    test "find the definition when call a erlang module", %{project: project} do
+      uses_content = ~q[
+        :erlang.binary_to_atom|("1")
+      ]
+      {:ok, uri, definition_line} = definition(project, uses_content)
+
+      assert uri =~ "/src/erlang.erl"
+      assert definition_line =~ ~S[«binary_to_atom»(Binary)]
+    end
+  end
+
+  defp caller_position(uses_content) do
+    {line, character} = cursor_position(uses_content)
+    Position.new(line, character)
+  end
+
+  defp definition(project, uses_content) do
+    position = caller_position(uses_content)
+
+    with {:ok, uses_file} <- open_uses_file(project, strip_cursor(uses_content)),
+         {:ok, {source_file, range}} <-
+           RemoteControl.Api.definition(project, uses_file, position),
+         {:ok, definition_line} <- definition_line(source_file, range) do
+      {:ok, source_file.uri, definition_line}
+    end
+  end
+
+  defp definition_line(source_file, range) do
+    with {:ok, line_text} <- SourceFile.fetch_text_at(source_file, range.start.line) do
+      graphemes = String.graphemes(line_text)
+      {text_before_range, range_text, text_after_range} = extract_range(graphemes, range)
+      marked_text = mark_range(range_text)
+      {:ok, IO.iodata_to_binary(text_before_range ++ marked_text ++ text_after_range)}
```

Comment:
use IOlists properly.
```elixir
marked_text = `IO.iodata_to_binary([text_before_range, mark_range(range_text), text_after_range])
{:ok, marked_text}
```
You don't need to build a single list (and you can use improper lists too). Concatenating lists together with `++` removes all the performance advantages to IOlists.